src/libraries/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.OSX.cs

```diff
@@ -40,10 +37,28 @@ private static ProcessInfo CreateProcessInfo(int pid)
             {
                 // Set the values we have; all the other values don't have meaning or don't exist on OSX
                 Interop.libproc.proc_taskallinfo temp = info.Value;
-                unsafe { procInfo.ProcessName = Marshal.PtrToStringAnsi(new IntPtr(temp.pbsd.pbi_comm))!; }
-                procInfo.BasePriority = temp.pbsd.pbi_nice;
-                procInfo.VirtualBytes = (long)temp.ptinfo.pti_virtual_size;
-                procInfo.WorkingSet = (long)temp.ptinfo.pti_resident_size;
+                string processName;
+                unsafe { processName = Marshal.PtrToStringAnsi(new IntPtr(temp.pbsd.pbi_comm))!; }
+                if (!string.IsNullOrEmpty(processNameFilter) && !string.Equals(processName, processNameFilter, StringComparison.OrdinalIgnoreCase))
+                {
+                    return null;
+                }
+
+                procInfo = new ProcessInfo()
+                {
+                    ProcessId = pid,
+                    ProcessName = processName,
+                    BasePriority = temp.pbsd.pbi_nice,
+                    VirtualBytes = (long)temp.ptinfo.pti_virtual_size,
+                    WorkingSet = (long)temp.ptinfo.pti_resident_size,
+                };
+            }
+            else
+            {
+                procInfo = new ProcessInfo()
```

Comment:
should these dummies be suppressed if there's a filter passed? that may be the cause of the test failure.